### ISSUE TITLE
TOML table arrays dropped when defined before parent object properties

### ISSUE THREAD

**Issue by cronik** [mikefarah/yq#1922]
**Describe the bug**
TOML table arrays dropped when defined before parent object properties. In otherwords, if you take the test case from [toml_test.go](https://github.com/mikefarah/yq/blob/33a3bb3bdfbf0ac5e6092e0314be70139c29ee19/pkg/yqlib/toml_test.go#L26-L38) and re-order the sections such the the "owner" section is after the "owner.addresses" sections the "addresses" field is dropped/deleted.

Version of yq: v4.40.5
Operating system: alpine linux
Installed via: binary

**Input Toml**

data1.toml:
```toml
[[owner.addresses]]
street = "first street"
suburb = "ok"

[[owner.addresses]]
street = "second street"
suburb = "nice"

[owner]
name = "Tom Preston-Werner"
age = 36
```

**Command**
The command you ran:
```
yq -oj .  data1.toml
```

**Actual behavior**

```json
{
  "owner": {
    "name": "Tom Preston-Werner",
    "age": 36
  }
}
```

**Expected behavior**

```json
{
  "owner": {
    "name": "Tom Preston-Werner",
    "age": 36,
    "addresses": [
      {
        "street": "first street",
        "suburb": "ok"
      },
      {
        "street": "second street",
        "suburb": "nice"
      }
    ]
  }
}
```

**Additional context**
The go-toml library seems to handle this as expected (see https://go.dev/play/p/D04-xTgNjx9 ), so I suspect there might be something in `decoder_toml.go`

**Comment by mikefarah** [comment]
Thanks for raising 👍🏼  will fix in the next release

**Comment by mikefarah** [comment]
Fixed in 4.41.1
